### PR TITLE
[DISCOVERY-92] fix consistency issue on deployments report

### PR DESCRIPTION
**Pull request by bruno-fs**
- Add common infrastructure for testing
- Add a container-based integration test for network scan
- Fix a consistency issue on deployments.json returned on distinct endpoints

### CODE REVIEW

File: quipucords/api/deployments_report/util.py [hunk 1]
```diff
@@ -83,7 +82,7 @@ def create_deployments_csv(deployments_report_dict, request):
     deployment_report_buffer = StringIO()
     csv_writer = csv.writer(deployment_report_buffer, delimiter=',')
 
-    systems_list = deployments_report_dict.get('system_fingerprints')
+    systems_list = deployments_report_dict.get("system_fingerprints")
```
Comment (nicolearagao) [inline quipucords/api/deployments_report/util.py:85]: why did you change single quotation marks for double on this specific example?

File: quipucords/tests/integration/test_network_scan.py [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+        "cpu_count": "cpu_count",
+        "cpu_socket_count": "cpu_socket_count",
+        "cpu_core_count": "cpu_core_count",
+        "cpu_core_per_socket": "cpu_core_per_socket",
+        "cpu_hyperthreading": "cpu_hyperthreading",
+        "system_last_checkin_date": "connection_timestamp",
+        "infrastructure_type": "virt_what_type/virt_type",
+        "system_creation_date": "date_machine_id",
+    }
+
```
Comment (nicolearagao) [inline quipucords/tests/integration/test_network_scan.py:130]: You wanted to cover all facts generated by the fingerprinter step or just a few of the most common ones ?

File: dev-requirements.txt [hunk 3]
```diff
@@ -219,6 +221,10 @@ darker==1.4.1 \
     --hash=sha256:c63cbfcf707987bc5cb3da30eaf5f2348aabc03f96b7b69bc31a34e96d8f2b8a \
     --hash=sha256:d423f4d791ff9a752a4a567a074d6c352cf6eb938d24d026a5799fcaa739c6c5
     # via -r dev-requirements.in
+distro==1.7.0 \
+    --hash=sha256:151aeccf60c216402932b52e40ee477a939f8d58898927378a02abbe852c1c39 \
```
Comment (nicolearagao) [inline dev-requirements.txt:225]: WHy the extra hashes here?

File: quipucords/tests/constants.py [hunk 4]
```diff
@@ -19,7 +19,7 @@
 QPC_COMMIT = "test-commit"
 QPC_SERVER_PASSWORD = "test-password"
 QPC_SERVER_USERNAME = "test-username"
-SCAN_TARGET_USERNAME = "non-root-user"
+READINESS_TIMEOUT_SECONDS = 60
```
Comment (nicolearagao) [inline quipucords/tests/constants.py:22]: Why did you have to change the order?

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -2,12 +2,15 @@ name: Test
...
-    branches: [ master ]
+    branches: [master]
   pull_request:
-    branches: [ master ]
+    branches: [master]
 
 jobs:
   test:
+    strategy:
+      matrix:
```
Comment (nicolearagao) [inline .github/workflows/test.yml:12]: I didn't know you could add the test matrix to github actions. This is cool.

**Review by nicolearagao — APPROVED** [review]
This looks pretty awesome. =)

File: quipucords/api/deployments_report/util.py (hunk 1 above)
Comment (bruno-fs) [reply quipucords/api/deployments_report/util.py:85]: this was black recommendation. I should probably have splited this in two separate commits.

Comment (bruno-fs) [reply quipucords/api/deployments_report/util.py:85]: and this line shouldn't be here. ouch. I moved it to line 98 :facepalm:

File: quipucords/tests/integration/test_network_scan.py (hunk 2 above)
Comment (bruno-fs) [reply quipucords/tests/integration/test_network_scan.py:130]: only the ones that are being collected.
This will require a refactor when all facts are collected

File: dev-requirements.txt (hunk 3 above)
Comment (bruno-fs) [reply dev-requirements.txt:225]: what do you mean by extra? :thinking: these are the hashes for `distro`

File: quipucords/tests/constants.py (hunk 4 above)
Comment (bruno-fs) [reply quipucords/tests/constants.py:22]: alfanumeric order lol. I'll change the commits to not have this juggling around

File: .github/workflows/test.yml (hunk 5 above)
Comment (bruno-fs) [reply .github/workflows/test.yml:12]: :)

File: quipucords/tests/utils/http.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    """Auth class for Quipucords server."""
+
+    def __init__(self, *, base_url, username, password):
+        """Initialize QPCAuth."""
+        self._qpc_client = ApiClient(base_url=base_url)
+        self._username = username
+        self._password = password
+
+    def __call__(self, r: requests.PreparedRequest):
+        """Add authorization token token to request headers."""
```
Comment (gtanzillo) [inline quipucords/tests/utils/http.py:50]: Minor: the word "token" appears twice

File: dev-requirements.txt (hunk 3 above)
Comment (nicolearagao) [reply dev-requirements.txt:225]: oh, I see

**Review by gtanzillo — APPROVED** [review]
